async_processor/sidecar/sidecar.py

```diff
@@ -10,21 +11,79 @@
     ProcessStatus,
 )
 from async_processor.logger import logger
-from async_processor.pydantic_v1 import BaseSettings, confloat
+from async_processor.pydantic_v1 import BaseModel, BaseSettings, confloat, conint
+from async_processor.sidecar.types import StreamBatch
 
 
-class Settings(BaseSettings):
+class _StreamConfig(BaseModel):
+    chunk_size: conint(ge=512, le=1024 * 512) = 1024 * 8
+
+
+class _Settings(BaseSettings):
     destination_url: str
     request_timeout: confloat(gt=0) = 3.0
+    stream_config: _StreamConfig = _StreamConfig()
 
     class Config:
         env_prefix = "TFY_ASYNC_PROCESSOR_SIDECAR_"
 
 
-settings = Settings()  # type: ignore
+settings = _Settings()  # type: ignore
+logger.info(repr(settings))
+
+
+class _StreamBatcher:
+    def __init__(self, stream_reader: aiohttp.StreamReader, config: _StreamConfig):
+        self._stream_reader = stream_reader
+        self._config = config
+        self._batch = StreamBatch(data="")
+        self._len = 0
+        self._decoder = codecs.getincrementaldecoder("utf8")()
+        self._fetcher_task = asyncio.create_task(self._fetcher())
+
+    async def _fetcher(self):
+        while True:
+            if self._len >= self._config.chunk_size:
+                await asyncio.sleep(0)
+                continue
+            data = await self._stream_reader.read(n=self._config.chunk_size)
+            self._len += len(data)
+            decoded_data = self._decoder.decode(data, self._stream_reader.at_eof())
+            # print(decoded_data)
+            # if not self._batch.data:
+            #     print("slow client")
+
+            self._batch.data += decoded_data
+
+            if self._stream_reader.at_eof():
+                self._batch.is_eof = True
+                break
+
+    def _reset_batch(self):
+        self._batch = StreamBatch(data="")
+        self._len = 0
+
+    async def iter(self) -> AsyncIterator[StreamBatch]:
+        while True:
```

Comment:
The dynamics between `_fetcher` and `iter` are hard to reason about. Add some comments to explain the expected behaviour.

Reply:
done